tests/Doctrine/Tests/DBAL/Connections/MasterSlaveConnectionTest.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Connections;
+
+use Doctrine\DBAL\Connections\MasterSlaveConnection;
+use Doctrine\DBAL\Driver;
+use Doctrine\Tests\DbalTestCase;
+
+class MasterSlaveConnectionTest extends DbalTestCase
+{
+    public function testConnectionParamsRemainAvailable(): void
+    {
+        $constructionParams = [
+            'driver' => 'pdo_mysql',
+            'keepSlave' => true,
+            'master' => [
+                'host' => 'master.host',
+                'user' => 'root',
+                'password' => 'password',
+                'port' => '1234',
+            ],
+            'slaves' => [
+                [
+                    'host' => 'slave1.host',
+                    'user' => 'root',
+                    'password' => 'password',
+                    'port' => '1234',
+                ],
+            ],
+        ];
+
+        $connection = new MasterSlaveConnection($constructionParams, $this->createStub(Driver::class));
+
+        $connectionParams = $connection->getParams();
+        foreach ($constructionParams as $key => $value) {
+            self::assertSame($value, $connectionParams[$key]);
+        }
```

Comment:
```suggestion
        self::assertEqualsCanonicalizing($constructionParams, $connection->getParams());
```

Reply:
the issue isn't the order, it's pollution of params by the constructor (`primary`, `replica`, `keepReplica`). Hence the loop. if you really want this, i'd have to `array_intersect` but i thought a loop was simpler.

Reply:
Oh right, nevermind then keep it that way 👍